File: Scripts/Script_passenger.py
import json

create_table = """
create table if not exists passenger
(
    passenger_id serial unique,
    id_number  varchar(50) not null primary key,
    name    varchar(50) not null,
    phone_number  varchar(50) not null,
    gender  varchar(5) not null,
    district   varchar(50) not null
);
"""

try:
    with open('../resource/passenger.json', 'r', encoding='utf-8') as file:
        data = json.load(file)
except Exception as e:
    print(f"Error reading or loading the JSON file: {e}")
    exit(1)

try:
    with open('../out/passenger.sql', 'w', encoding='utf-8') as file:
        file.write(create_table + "\n")
        for record in data:
            id_number = record['id_number'].replace("'", "''")
            name = record['name'].replace("'", "''")
            phone_number = record['phone_number'].replace("'", "''")
            gender = record['gender'].replace("'", "''")
            district = record['district'].replace("'", "''")

            file.write(f"INSERT INTO passenger (id_number, name, phone_number, gender, district) "
                       f"VALUES ('{id_number}', '{name}', '{phone_number}', '{gender}', '{district}');\n")
except Exception as e:
    print(f"Error writing to the SQL file: {e}")

File: Scripts_in_Java/Script_passenger.java
import org.json.JSONArray;
import org.json.JSONObject;

import java.io.*;
import java.nio.charset.StandardCharsets;

public class Script_passenger {

    public static void main(String[] args) {
        String createTableSql = """
                create table if not exists passenger
                (
                    passenger_id serial unique,
                    id_number varchar(50) not null primary key,
                    name varchar(50) not null,0
                    phone_number varchar(50) not null,
                    gender varchar(5) not null,
                    district varchar(50) not null
                );
                """;

        String jsonFilePath = "C:\\Users\\Eric\\IdeaProjects\\CS307\\Project1\\resource\\passenger.json";
        String sqlFilePath = "C:\\Users\\Eric\\IdeaProjects\\CS307\\Project1\\out\\passenger.sql";

        try (BufferedReader reader = new BufferedReader(new FileReader(jsonFilePath, StandardCharsets.UTF_8))) {
            StringBuilder contentBuilder = new StringBuilder();
            String line;
            while ((line = reader.readLine()) != null) {
                contentBuilder.append(line);
            }
            String jsonData = contentBuilder.toString();
            JSONArray data = new JSONArray(jsonData);

            try (BufferedWriter writer = new BufferedWriter(new FileWriter(sqlFilePath, StandardCharsets.UTF_8))) {
                writer.write(createTableSql);
                for (int i = 0; i < data.length(); i++) {
                    JSONObject record = data.getJSONObject(i);
                    String idNumber = record.getString("id_number").replace("'", "''");
                    String name = record.getString("name").replace("'", "''");
                    String phoneNumber = record.getString("phone_number").replace("'", "''");
                    String gender = record.getString("gender").replace("'", "''");
                    String district = record.getString("district").replace("'", "''");

                    String insertSql = String.format("INSERT INTO passenger (id_number, name, phone_number, gender, district) VALUES ('%s', '%s', '%s', '%s', '%s');\n",
                            idNumber, name, phoneNumber, gender, district);
                    writer.write(insertSql);
                }
            }
        } catch (Exception e) {
            System.out.println("Error: " + e.getMessage());
        }
    }
}
